sources/packages/api/src/route-controllers/institution-locations/institution-locations.controller.ts

```diff
@@ -111,4 +136,24 @@ export class InstitutionLocationsController extends BaseController {
     );
     return Institutionlocations;
   }
+
+  @Get(":locationId")
+  async getInstitutionLocation(@Param("locationId") locationId: number,
+    @UserToken() userToken: IUserToken,
+  ): Promise<InstitutionLocationsDetailsDto> {
+    //To retrive institution id
+    const institutionDetails = await this.institutionService.getInstituteByUserName(
+      userToken.userName,
+    );
+    if (!institutionDetails) {
+      throw new UnprocessableEntityException("Not able to find the Location associated.");
+    }
+    // get all institution locations.
+    const Institutionlocations = await this.locationService.getInstitutionLocation(
+      institutionDetails.id,
```

Comment:
It seems that it will ensure that the location being requested belongs to the institution but the code is not validating if the user has access to the specific location.
we can add the validation or create a tech debit to load into the userToken an array of all locations that the user has access to and validate based on it.

Reply:
Created #476 to handle it.